test/e2e/secondary_network_test.go

```diff
@@ -0,0 +1,286 @@
+// Copyright 2022 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+	"time"
+
+	logs "github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v3"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+// Structure to extract and store the secondary network configuration information parsed from secondary_network_config_param.yml.
+type PodConfig struct {
+	InterfaceType struct {
+		InterfaceType string `yaml:"interfacetype"`
+	} `yaml:"interface_type"`
+	SriovConf struct {
+		Networkinterface string `yaml:"networkinterface"`
+		Numberofvfs      int    `yaml:"numberofvfs"`
+	} `yaml:"sriov_conf"`
+	VirNet struct {
+		TotalNumberOfVirtualNetworks int      `yaml:"totalnumberofvirtualnetworks"`
+		Virtualnetworknames          []string `yaml:"virtualnetworknames"`
+	} `yaml:"vir_net"`
+	CreatePod struct {
+		Numberofpods int             `yaml:"numberofpods"`
+		Describe     [][]interface{} `yaml:"describe"`
+	} `yaml:"create_pod"`
+}
+
+var service PodConfig
+
+// Structure for extracting the variables for describing the pod from secondary_network_config_param.yaml file.
+type describePodInfo struct {
+	nameOfPods                   string
+	countOfVirtualNetworksPerPod int
+	nameOfVirtualNetworkPerPod   []string
+	nameOfInterfacePerPod        []string
+}
+
+var podData []describePodInfo
+var totalNumberOfPods int
+var interfaceType string
+
+const secondary_network_config_yaml = "./infra/secondary_network/secondary_network_configuration.yml"
+const nameSpace = "kube-system"
+const ctrName = "busyboxpod"
+const osType = "linux"
+const count = 5
+const size = 40
+const reqName = "intel.com/intel_sriov_netdevice"
+const resNum = 3
+const (
+	podName = iota
+	podVNsCount
+	podVirtualNetwork
+	podInterfaceName
+)
+
+// The function ensureAntreaRunningWithSecondaryNetworkOption ensures that the antrea is enabled and running with sriov and whereabouts secondary network options
+func (data *TestData) ensureAntreaRunningWithSecondaryNetworkOption() error {
```

Comment:
IMO this doesn't belong in the test files. The scripts in https://github.com/antrea-io/antrea/pull/4097 already take care (or should take care) of deploying Antrea with the correct options for SecondaryNetwork testing.

Based on our Slack discussion, this is to counteract https://github.com/antrea-io/antrea/blob/e7486d59528c128b24d03be3fb4a177269eb77a3/test/e2e/main_test.go#L122 

from the main function, which overwrites the Antrea deployment.

Which means that:
- we have scripts that deploy Antrea with the correct configuration
- the deployment is overwritten by `ensureAntreaRunning`
- we have to run `ensureAntreaRunningWithSecondaryNetworkOption` to go back to the correct depoloyment

That seems wasteful.

I think the best option to move forward is to move these tests to `test/e2e-secondary-network/` and let them have their own `main` function. I still feel that these tests don't really fit well within the main suite of tests, which supports things like code coverage measurement.